packages/lesswrong/components/hooks/useUpdateGlossaryPinnedState.ts

```diff
@@ -0,0 +1,30 @@
+import { useUpdate } from "@/lib/crud/withUpdate";
+import { useCurrentUser } from "../common/withUser";
+import { useCallback } from "react";
+
+export function useGlossaryPinnedState() {
+  const currentUser = useCurrentUser();
+  const {mutate: updateUser} = useUpdate({
+    collectionName: "Users",
+    fragmentName: 'UsersCurrent',
+  });
+  
+  const togglePin = useCallback(async () => {
+    // TODO: figure out what to do for logged out users... should it just be local state individualized by post, or pop up a login modal?
```

Comment:
This seems like it still needs to be figured out

Reply:
We decided to disable click-behavior for the glossary for logged out users for the MVP to ship, so they won't be able to toggle it.  Left the TODO in as a reminder to come back to it if we decide we want to enable toggling locally, or something.